src/app/Services/UserService.php

```diff
@@ -75,4 +78,25 @@ public function resetPassword($token, $newPassword): User
 
         return $user;
     }
+
+    /**
+     * Get User Profile
+     */
+
+     public function getUserProfile(int $id): array
+     {
+         try {
+
+             $user = User::with('profile')->findOrFail($id);
+             $posts = Post::where('user_id', $user->id)->get();
+     
+             return compact('user', 'posts');
+         } catch (ModelNotFoundException $e) {
+             return ['error' => 'User not found.'];
+         } catch (QueryException $e) {
+             return ['error' => 'Database error: ' . $e->getMessage()];
+         } catch (\Exception $e) {
+             return ['error' => 'An unexpected error occurred: ' . $e->getMessage()];
+         }
```

Comment:
just one catch is enough, `$e` already contains the information about the error

Reply:
Done